tests/remote_resource/test_requests.py

```diff
@@ -0,0 +1,257 @@
+        self.assertEqual(resources[0].id, '1')
+        self.assertEqual(resources[0].title, 'Omakase')
+        self.assertEqual(resources[0].author.type, 'People')
+        self.assertEqual(resources[0].author.id, '9')
+
+    def test_remote_resource_list_wrapper_multiple_relationship(self):
+        data = {
+            "data": [
+                {
+                    "type": "articles",
+                    "id": "1",
+                    "attributes": {
+                        "title": "Omakase"
+                    },
+                    "relationships": {
+                        "authors": {
+                            "links": {
+                                "self": "/articles/1/relationships/authors",
+                                "related": "/articles/1/authors"
+                            },
+                            "data": [
+                                {"type": "People", "id": "9"},
+                            ]
+                        }
+                    }
+                }
+            ]
+        }
+
+        resources = RemoteResourceListWrapper(data['data'])
+
+        self.assertEqual(resources[0].type, 'articles')
+        self.assertEqual(resources[0].id, '1')
+        self.assertEqual(resources[0].title, 'Omakase')
+        self.assertEqual(resources[0].authors[0].type, 'People')
+        self.assertEqual(resources[0].authors[0].id, '9')
+
+
+@mock.patch('zc_common.remote_resource.request.requests')
+@mock.patch('zc_common.remote_resource.request.zc_settings')
+class RouteRetrievalTestCase(TestCase):
+
+    def setUp(self):
+        self.gateway_root_path = 'http://mp-gateway.zerocater.com'
+        self.mappings = {
+            '/users/{id}': {
+                'domain': 'https://mp-users.zerocater.com',
+                'resource_type': 'User'
+            }
+        }
+        self.get_response = mock.Mock()
+        self.get_response.json.return_value = self.mappings
+
+    def bind_mock_objects(self, mock_zc_settings, mock_requests):
+        mock_zc_settings.GATEWAY_ROOT_PATH = self.gateway_root_path
+        mock_requests.get.return_value = self.get_response
+
+    def test_wrong_resource_type(self, mock_zc_settings, mock_requests):
+        self.bind_mock_objects(mock_zc_settings, mock_requests)
+        resource_type = 'Movie'
+        self.assertRaises(Exception, get_route_from_fk, resource_type)
+
+    def test_correct_resource_type(self, mock_zc_settings, mock_requests):
+        self.bind_mock_objects(mock_zc_settings, mock_requests)
+        expected_url = 'https://mp-users.zerocater.com/users/1'
+
+        result_url = get_route_from_fk('User', {'id': 1})
+
+        mock_requests.get.assert_called_once_with(self.gateway_root_path)
+        self.assertEqual(expected_url, result_url)
+
+    def test_missing_required_parameters(self, mock_zc_settings, mock_requests):
+        resource_type = 'User'
+        self.assertRaises(Exception, get_route_from_fk, resource_type)
```

Comment:
For this situation, is this the behavior we want? I can't imagine a situation where a service would want to get the bare list collection (in this case, `https://mp-users.zerocater.com/users`), but it looks like the function doesn't support it. Not necessarily anything to change, but just want to make sure that's what's we want.